### PR TITLE
[BACK-145] Optimize fetching /users/:userid/users

### PR DESCRIPTION
**Pull request by pazaan**
`/users/:userid/users` was previously calling
`gatekeeperClient.getUserInfo` for every user in the associated user
list. For clinics, this could be a very large number of server-to-server
calls.
Code has now been changed to take advantage of `shoreline`'s new
`/users?id=userid1,userid2...` query parameter.

Also:
* Updated NPM packges to latest
* Removed downgrading mongojs package for Docker builds
* Migrate to yarn

Fixes:
* [BACK-145]
* [BACK-147]

[BACK-145]: https://tidepool.atlassian.net/browse/BACK-145
[BACK-147]: https://tidepool.atlassian.net/browse/BACK-147

### CODE REVIEW

File: lib/routes/seagullApi.js [hunk 1]
```diff
@@ -331,7 +331,11 @@ module.exports = function (crudHandler, userApiClient, gatekeeperClient, metrics
               user.profile = _.result(document, 'detail.profile');
 
               if (_.isEmpty(user.trustorPermissions)) {
-                delete user.profile.patient;
+                if (user.profile) {
+                  delete user.profile.patient;
+                } else {
+                  log.error(`User ${user.userid} does not have a valid profile. Consider investigaing the account.`);
```
Comment (tjotala) [inline lib/routes/seagullApi.js]: nit: typo 'investigaing'

Comment (pazaan) [reply lib/routes/seagullApi.js]: Gah! Why isn't there a normal spell check in VSCode :p

Comment (pazaan) [reply lib/routes/seagullApi.js]: Added one :)

File: lib/routes/seagullApi.js [hunk 2]
```diff
@@ -283,54 +281,79 @@ module.exports = function (crudHandler, userApiClient, gatekeeperClient, metrics
...
+                throw new Error(`Error getting users: ${error}`);
               }
+            });
+            userProfiles = _.flatten(results);
 
-              if (!userMatchesQueryOnUser(user, query)) {
-                return callback();
-              }
+            if (!_.isEqual(_.map(userProfiles, 'userid').sort(), _.sortBy(userIds))) {
+              throw new Error('Received less results than we queried for');
```
Comment (derrickburns) [inline lib/routes/seagullApi.js]: less => fewer :)

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "seagull",
-  "version": "0.9.0",
+  "version": "0.10.0-rc.1",
```
Comment (derrickburns) [inline package.json]: Is this what you want?

Comment (pazaan) [reply package.json]: I definitely want to go from `0.9.0` to `0.10.0`. I'm just prepping for the release candidate 😉

File: lib/routes/seagullApi.js [hunk 4]
```diff
@@ -281,54 +281,75 @@ module.exports = function (crudHandler, userApiClient, gatekeeperClient, metrics
...
+              } catch (error) {
+                throw new Error(`Error getting users: ${error}`);
               }
+            });
+            userProfiles = _.flatten(results);
 
-              if (!userMatchesQueryOnUser(user, query)) {
-                return callback();
-              }
+            if (!_.isEqual(_.map(userProfiles, 'userid').sort(), _.sortBy(userIds))) {
```
Comment (tjotala) [inline lib/routes/seagullApi.js]: Just curious: assuming no duplicates, wouldn't a simple length check suffice here?

Comment (pazaan) [reply lib/routes/seagullApi.js]: You're quite right.

**Pull request merged**